ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -95,21 +83,41 @@ export default ({
 
         if (newNode.type === "batch") {
           parentIdMapArray.push({ prevId: n.id, newId: newNode.id });
-        } else if (newNode.type === "subworkflow") {
+
+          const batchNodeChildren = nodes.filter(
+            (child) => child.parentId === n.id,
+          );
+
+          batchNodeChildren.forEach((child) => {
+            if (!pastedNodes.some((node) => node.id === child.id)) {
+              const childNewNode = {
+                ...child,
+                id: generateUUID(),
+                position: { ...child.position },
+                selected: true,
+                data: { ...child.data },
```

Comment:
You are already passing `...child`, which will come with data, so unless you are adding, or modifying something in data, you don't need this.
Same for position